Services/Membership/classes/class.ilParticipants.php

```diff
@@ -823,6 +823,12 @@ public function delete($a_usr_id)
         $rbacadmin = $DIC['rbacadmin'];
         $ilDB = $DIC['ilDB'];
 
+        if (!$this->checkLastAdmin([$a_usr_id])) {
+            $DIC->language()->loadLanguageModule($this->getType());
+            ilUtil::sendFailure($this->lng->txt($this->getType() . '_at_least_one_admin'), true);
```

Comment:
This introduces a dependency to the global template interface (trunk ilUtil::sendInfo replacement) in a non GUI-class.
Throwing an exception would be more appropiate here. 

Nevertheless even if an exception would be thrown here, I wouldn't introduce it in R7, because all consumers of ilParticipants:delete would have to be adapted/checked whether they handle the exception correctly.  E.g:

- SOAP calls with ilCourse|GroupXMLParser
- ECS member de/assignments have to be adapted. Otherwise they will fail. 
- maybe more

Reply:
Totally agree on that one, i missed that sideeffect in my concept.

Therefore i abstracted the response so the Error can be handled inside the GUI due to a return value of the progress